### ISSUE TITLE
End inline code with forward slash error

### ISSUE THREAD

**Issue by kescobo** [tlienart/Franklin.jl#807]
I suspect this is known, but I couldn't find a related issue - there's some weird behavior with inline code quoting ending with `\`. The following formulations all have some variant of `ERROR: I found the opening token 'CODE_SINGLE' but not the corresponding closing token.`

```
- Some inline code with `trailing \`
- Also no joy with double ``backticks \``
```

Interestingly, escaping the backslash works (eg ```\\```), but then both slashes show up. I discovered that doing inline triple backticks ~~actually does work.~~ does some weird things. That is, the following does not throw an error:

````
This is ```fine! \```
````

But depending on what it's surrounded by, can eliminate whole paragraphs, or just work with some extra backticks included. Still working on some MWE.

I was moving a post over from Hugo, and it took me a long time to find the problem, because I had ```\``` early in the post, but the stack trace kept referring to the last inline code in the post (I think I understand why, but annoying).

EDIT: The triple backticks is not the solution I first thought it was. Still working on some examples

**Comment by kescobo** [comment]
````
# This is a new blog post

- Then "et al.": ` et al\.`
  - the `.` is a special character, so it's escaped with ```\```
  - this is also optional, so I wrap it in parentheses and add `?`
- The year, which is 4 digits, and wrapped in parentheses: `\((\d{4})\)`

```julia
ismatch(r"\\", "some julia string with \\ ",)
```

This is ```fine! \```

This is gone?!?

```julia
println("some more julia code")
```

But how about this! ```\```

And this?

This is ```NOT fine! \``` (it errors if at the end)

```julia
"Line above renders if this code block is present"
```
````

![image](https://user-images.githubusercontent.com/3502975/113580504-d9e22480-95f3-11eb-952d-7b175145aa11.png)

**Comment by tlienart** [comment]
Hello!

The issue here is that `\` is a special character since we allow latex-style commands. It also serves as an escaper for instance `\$` for the dollar symbol (which otherwise is also a special character) or indeed `` \` `` which escapes the backtick so it can be shown as a single character.

Now let's inspect the following and hopefully clarify what's going on so you know what to do:

````julia
julia> println(fd2html(raw"`\ `"))
<code>\</code>
````

This is what I think you want, note that the whitespace is significant here to disambiguate between escaping the backtick and closing the inline code block. Note that this whitespace gets stripped in the generated HTML.

So in short, in code, don't use backslashes at the end of the code without adding a whitespace to disambiguate with the closing backticks.
Does that answer your question?

**Comment by kescobo** [comment]
>Note that this whitespace gets stripped in the generated HTML.

Neat - would not have guessed that! Explanation makes sense, I assumed it was something like that - glad to know there's a solution :+1: 

PR made #809

**Comment by tlienart** [comment]
Thanks a lot!
